pkg/registry/certificates/certificates/strategy.go

```diff
@@ -190,3 +194,48 @@ func (csrApprovalStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 func (csrApprovalStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateCertificateSigningRequestUpdate(obj.(*certificates.CertificateSigningRequest), old.(*certificates.CertificateSigningRequest))
 }
+
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
+	csr, ok := obj.(*certificates.CertificateSigningRequest)
+	if !ok {
+		return nil, nil, fmt.Errorf("not a certificatesigningrequest")
+	}
+	return labels.Set(csr.Labels), SelectableFields(csr), nil
+}
+
+// MatchCSR returns a generic matcher for a given label and field selector.
+func MatchCSR(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
+	return storage.SelectionPredicate{
+		Label:       label,
+		Field:       field,
+		GetAttrs:    GetAttrs,
+		IndexFields: []string{"spec.signerName"},
+	}
+}
+
+// NameTriggerFunc returns value metadata.namespace of given object.
+func NameTriggerFunc(obj runtime.Object) string {
+	return obj.(*certificates.CertificateSigningRequest).ObjectMeta.Name
+}
+
+// SignerNameTriggerFunc returns value spec.signerName of given object.
+func SignerNameTriggerFunc(obj runtime.Object) string {
+	return valueOrEmpty(obj.(*certificates.CertificateSigningRequest).Spec.SignerName)
+}
+
+// SelectableFields returns a field set that can be used for filter selection
+func SelectableFields(obj *certificates.CertificateSigningRequest) fields.Set {
+	objectMetaFieldsSet := generic.ObjectMetaFieldsSet(&obj.ObjectMeta, true)
+	csrSpecificFieldsSet := fields.Set{
+		"spec.signerName": valueOrEmpty(obj.Spec.SignerName),
```

Comment:
If the pointer is nil, we set this to "" (the empty string value). I am fairly certain this is correct.. 😄

Reply:
Yeah, that looks reasonable.